src/components/my-plans/search-panel/search-panel.js

```diff
@@ -1,27 +1,76 @@
 import React, { Component } from 'react';
+import Api from '../../../services/places-service';
 import './search-panel.scss';
 
 export default class SearchPanel extends Component {
-  onCommentChange = (event) => {
-    this.setState({ comment: event.target.value });
+  state = {
+    query: [],
+    inputValue: ''
+  }
+  onCommentChange = (e) => {
+    const inputValue = e.target.value;
+    clearTimeout(this.timeout)
+    this.setState({
+      inputValue: e.target.value
+    })
+    if (inputValue.length >= 1 && inputValue.trim()) {
+      this.timeout = setTimeout(() => {
+        Api.getSearchParameters(inputValue).then((query) => {
+          if (query.length > 0) {
+            this.setState({ query })
+          }
+          else {
+            query.push({ name: 'There is no such a place' });
+            this.setState({ query })
+          }
+        }
+        );
+      }, 400)
+    }
+    else {
+      this.setState({ query: [] })
+    }
   };
+
   onSubmit = (event) => {
     event.preventDefault();
-    this.props.onButtonAddClick(this.state.comment)
+    this.props.onButtonAddClick(this.state.inputValue)
+    this.setState({
+      inputValue: ''
+    })
   };
+
   render() {
+    const options = this.state.query.map(place => (
+      <li key={place._id}
+        onClick={() => {
+          this.setState({
+            inputValue: place.name
+          })
+          this.setState({ query: [] })
```

Comment:
You can set several states at one setState function. `setState({ inputValue: place.name, query: [] })`

Reply:
done